### ISSUE TITLE
[SUGGESTION] Upgrade Framework dependencies

### ISSUE THREAD

**Issue by JKamsker** [mbdavid/LiteDB#2510]
This issue is just for keeping track of why we should upgrade the dependencies of the project from ``net4.5;netstandard1.3;netstandard2.0`` to newer versions.

- Microsoft officially doesnt support part of the frameworks anymore
- Span<T> support, tho it can be fixed using the ``System.Memory`` package

Open Questions:
- Do polyfills affect newer versions that have the required apis already implemented? (will the polyfill be present in those cases?)

Related:
* https://github.com/mbdavid/LiteDB/pull/2508

**Comment by ngoquoctoandev** [comment]
I also agree with your point of view. In an ever-evolving and constantly changing technological world, I think targeting the latest development frameworks would be more appropriate. For example, .NET 8

**Comment by Joy-less** [comment]
[This pull request](https://github.com/litedb-org/LiteDB/pull/2564) will replace support for `.NET 4.5` and `.NET Standard 1.3` with support for `.NET Standard 2.0`, `.NET Standard 2.1`, `.NET 8.0` and `.NET 9.0` if it is merged. But it's possible that framework changes will only happen in LiteDB v6+.
